### ISSUE TITLE
Missing tms or teamId key leads to fatal server-side error

### ISSUE THREAD

**Issue by jordanh** [ParabolInc/action#1810]
## Issue - Bug

When we restore the latest production database to staging and test, we end up using development/staging auth0 users on top of production data. This will cause some data synchronization issues, most notably that the `tms` field of the token will be out of sync with the database.

At present, when there are keys in `tms` that don't exist in the DB, we see the following exception on the server:

```
2018-02-28T17:48:38.623264064Z app[web.1]: DEBUG GraphQL Error: [ { Error: Key not found : rJtg8fTyM
2018-02-28T17:48:38.623304474Z app[web.1]:     at /app/src/server/utils/RethinkDataLoader.js:18:67
2018-02-28T17:48:38.623321429Z app[web.1]:     at Array.map (<anonymous>)
2018-02-28T17:48:38.623328182Z app[web.1]:     at /app/src/server/utils/RethinkDataLoader.js:18:15
2018-02-28T17:48:38.623333756Z app[web.1]:     at _callee$ (/app/src/server/utils/RethinkDataLoader.js:26:12)
2018-02-28T17:48:38.623339354Z app[web.1]:     at tryCatch (/app/node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:65:40)
2018-02-28T17:48:38.623361066Z app[web.1]:     at Generator.invoke [as _invoke] (/app/node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:303:22)
2018-02-28T17:48:38.623367187Z app[web.1]:     at Generator.prototype.(anonymous function) [as next] (/app/node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:117:21)
2018-02-28T17:48:38.623373157Z app[web.1]:     at step (/app/src/server/utils/RethinkDataLoader.js:24:191)
2018-02-28T17:48:38.623378627Z app[web.1]:     at /app/src/server/utils/RethinkDataLoader.js:24:361
2018-02-28T17:48:38.623383959Z app[web.1]:     at <anonymous>
2018-02-28T17:48:38.623389442Z app[web.1]:     message: 'Key not found : rJtg8fTyM',
2018-02-28T17:48:38.623394659Z app[web.1]:     locations: [ [Object], [Object] ],
2018-02-28T17:48:38.623400044Z app[web.1]:     path: [ 'viewer', 'teams' ] } ]
```

This is a bummer because it effectively prevents us from testing with production data.

This also shows up when a `MENTION` notification exists with a `teamId` field set for a team that no longer exists. N.B. we should likely scrutinize the delete team mutation to make sure it cleans up notifications _and_ make the Notifications container a bit more robust.

- **OS:** macOS 
- **Browser:** Chrome
- **Estimated effort:** 12 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

### Acceptance Criteria (optional)
A tester can:
   - Log in to staging with an id that is the same as their id on production
   - If there is a `tms` mismatch, the DB's fields are used as the single-source of truth and,
   - The `tms` field is updated by the server

**Comment by mattkrick** [comment]
oh boy, this is a fantastic "working as intended" scenario.

if there is a mismatch, we send the client a new token, but probably not before they fire off the query. 
that's because if we waited for verification, then the web socket server would either not have a listener set up yet, or it would think that the request was unauthorized.

To fix this, we have 2 options:
- set up a queue on the server that holds onto client messages until the auth is complete. we'd need to be smart & flush this after 1 or 2 seconds to avoid a memory leak as that'd be an easy attack vector.
- on the client, don't send a request until an ACK is received from the server. this is safer, but means data will take an extra ~300ms to get to the client.

 as an alternative, we could gracefully catch teamIds that don't exist in the DB.

**Comment by jordanh** [comment]
I like the third option, myself. Fast, and would facilitate testing.

**Comment by mattkrick** [comment]
so before we build that could affect production for the purposes of testing, can i ask why there's a mismatch? were you still logged in on staging?

**Comment by jordanh** [comment]
I've used my production user (logging in with auth0 via Google) to log in on staging since the beginning of time. Over the years (years!) that `tms` key has got pretty crufty :)

Let your conscious be your guide (as always) but no need to fix this one up until we prioritize it. Right now, I think we can just let it hang.

**Comment by jordanh** [comment]
This also shows up when there is a Notification for a team that no longer exists. Updating the acceptance criteria above...

**Comment by jordanh** [comment]
See: #1774

**Comment by jordanh** [comment]
Addressed by #1868
